diff --git a/stand/efi/loader/main.c b/stand/efi/loader/main.c
index 5edc06c9e0d..773f699158a 100644
--- a/stand/efi/loader/main.c
+++ b/stand/efi/loader/main.c
@@ -961,7 +961,7 @@ main(int argc, CHAR16 *argv[])
 	 */
 	setenv("console", "efi", 1);
 	uhowto = parse_uefi_con_out();
-#if defined(__riscv)
+#if defined(__aarch64__) || defined(__arm__)  || defined(__riscv)
 	/*
 	 * This workaround likely is papering over a real issue
 	 */
