diff --git a/stand/efi/loader/main.c b/stand/efi/loader/main.c
index 7b78f94eb368..73936d870771 100644
--- a/stand/efi/loader/main.c
+++ b/stand/efi/loader/main.c
@@ -947,7 +947,7 @@ main(int argc, CHAR16 *argv[])
 	 */
 	setenv("console", "efi", 1);
 	uhowto = parse_uefi_con_out();
-#if defined(__riscv)
+#if defined(__aarch64__) || defined(__arm__) || defined(__riscv)
 	if ((uhowto & RB_SERIAL) != 0)
 		setenv("console", "comconsole", 1);
 #endif
