diff --git a/sys/modules/dtb/rockchip/Makefile b/sys/modules/dtb/rockchip/Makefile
index 1e0c6a2d436..c17a6b11895 100644
--- a/sys/modules/dtb/rockchip/Makefile
+++ b/sys/modules/dtb/rockchip/Makefile
@@ -17,7 +17,8 @@ DTS=	\
 	rockchip/rk3328-rock-pi-e.dts \
 	rockchip/rk3328-rock64.dts \
 	rockchip/rk3399-firefly.dts \
-	rockchip/rk3399-rockpro64.dts
+	rockchip/rk3399-rockpro64.dts \
+	rockchip/rk3399-pinebook-pro.dts
 
 DTSO=	rk3328-analog-sound.dtso \
 	rk3328-i2c0.dtso \
