diff --git a/sys/arm64/conf/GENERIC-DRM b/sys/arm64/conf/GENERIC-DRM
index 7243a4f35b9..2ca1b117bd5 100644
--- a/sys/arm64/conf/GENERIC-DRM
+++ b/sys/arm64/conf/GENERIC-DRM
@@ -18,8 +18,10 @@ device		drm
 device		aw_de2_drm
 device		rk_drm
 device		dw_hdmi
+device		panfrost
 
 files		"../../dev/drm/extra_patches/files"
+files		"../../dev/drm/extra_patches/files.arm64"
 files		"../../dev/drm/extra_patches/files.aw"
 files		"../../dev/drm/extra_patches/files.bridges"
 files		"../../dev/drm/extra_patches/files.rk"
diff --git a/sys/conf/files.arm64 b/sys/conf/files.arm64
index f793bf9863a..a21515e0a11 100644
--- a/sys/conf/files.arm64
+++ b/sys/conf/files.arm64
@@ -104,7 +104,7 @@ arm64/coresight/coresight_tmc_fdt.c		optional fdt
 
 arm64/iommu/iommu.c				optional iommu
 arm64/iommu/iommu_if.m				optional iommu
-arm64/iommu/iommu_pmap.c			optional iommu
+arm64/iommu/iommu_pmap.c			optional iommu | panfrost
 arm64/iommu/smmu.c				optional iommu
 arm64/iommu/smmu_acpi.c				optional iommu acpi
 arm64/iommu/smmu_fdt.c				optional iommu fdt
